test/integration/proxy_req_test.js

```diff
@@ -358,3 +358,122 @@ test('overrides /proxy/req endpoint and fails', function t(assert) {
         });
     });
 });
+
+test('aborts retry because keys diverge', function t(assert) {
+    assert.plan(4);
+
+    var cluster = allocCluster(function onReady() {
+        // Make node two refuse initial request
+        cluster.two.membership.checksum = cluster.one.membership.checksum + 1;
+
+        cluster.one.on('requestProxy.retryAborted', function onRetryAborted() {
+            assert.pass('retry aborted');
+        });
+
+        cluster.one.on('requestProxy.retryScheduled', function onRetryScheduled() {
+            // Make sure keys diverge
+            cluster.one.lookup = lookupHandler();
+
+            function lookupHandler() {
+                var count = 0;
+
+                return function lookIt() {
+                    return ++count === 1 ? cluster.one.hostPort :
+                        cluster.three.hostPort;
+                };
+            }
+        });
+
+        cluster.requestAll({
+            keys: [cluster.keys.two, cluster.keys.two],
+            host: 'one',
+            maxRetries: 10,
```

Comment:
is it possible to have an `assert.equal(retryCount, 0)` ? 

We want to test early abort but it doesnt look like we have an assert for it.